lib/replace-selection-items.js

```diff
@@ -0,0 +1,14 @@
+import {splice} from 'immutable-array-methods';
+import getSelectionItems from './get-selection-items';
+
+export default (articleJson, fn) => {
+  const items = getSelectionItems(articleJson);
+
+  if (!items.length) {
+    return articleJson;
+  }
```

Comment:
oh I see now - if we've got `items.length === 0` we don't want to call `fn(items)` further down. So this is indeed needed :)